### PR TITLE
TFLite: Added support for ZeroPadding3D, AveragePooling3D & MaxPooling3D in OpenCL

### PR DESCRIPTION
**Pull request by guy-david**
A step towards solving [this](https://github.com/tensorflow/tensorflow/issues/34010) issue.
This pull request was sent to get early feedback from the dev team.

### CODE REVIEW

**Comment by guydavid** [comment]
I have just realized that my changes to pad.cc and pad_test.cc are not required and better be reverted. However, I originally made them because when you enable the GPU delegate, it first prepares the CPU kernels [here](https://github.com/guydavid/tensorflow/blob/lite5d/tensorflow/lite/core/subgraph.cc#L1144). Can you explain this step?

File: tensorflow/lite/core/subgraph.cc [hunk 1]
```diff
@@ -1141,20 +1141,24 @@ TfLiteStatus Subgraph::ModifyGraphWithDelegate(TfLiteDelegate* delegate) {
...
-      // Make sure that we are in a defined ready state before returning.
-      // Plan and allocate tensors before returning.
-      TF_LITE_ENSURE_OK(&context_, EnsureMemoryAllocations());
-      ReportError(
-          "Attempting to use a delegate that only supports static-sized "
-          "tensors with a graph that has dynamic-sized tensors.");
-      return kTfLiteError;
-    }
-  }
+  // TODO(guydavid): The following is commented because it makes use of the
```
Comment (impjdi) [inline tensorflow/lite/core/subgraph.cc]: I don't think you can just comment this out like this...

**Review by impjdi — COMMENTED** [review]
I haven't read the entire thread, but based on the title of this pull request, you're introducing new ops: zero padding 3d, average pooling 3d, max pooling 3d.  in that case, I would prefer the following:

1. First doing the plumbing for the cpu (introducing new op code, add reference impl), so that we have a well-defined op and that people can use even without the gpu.
2. And then proceeding with the gpu impl in this PR.

But please sync with @jdduke too as he's the owner on TFLite side.

File: tensorflow/lite/core/subgraph.cc (hunk 1 above)
Comment (guydavid) [reply tensorflow/lite/core/subgraph.cc]: This was a temporary change and will be ideally resolved when I implement the reference implementation, per your advice.

**Comment by gbaned** [comment]
@guydavid Can you please resolve conflicts? Thanks!

**Comment by guydavid** [comment]
I have completely resolved the original issue linked in the description here with full OpenCL support. However, I don't see it going through.
There are some adjustments to be made to keep compatibility with older TF-Lite schema's as the new one has completely changed. Seeing the zero progress with other one-liner pull requests that I have made (this pull request has 5000+ changed lines) does not add incentives so I will be dropping this for now.

**Comment by jdduke** [comment]
@guydavid can you link to the other PRs? We're in the process of restructuring how we assign/prioritize PRs, so apologies if anything has fallen through the cracks. Happy to help push those through.

For this specific case, it's not that we don't want this, but adding a new op is fairly non-trivial at the moment (particularly as we're working on a new converter), and generally prefer having the conversion/CPU path in place before we extend it to accelerators.

**Pull request closed without merging**